### PR TITLE
refactor: navbar to typescript.

### PR DESCRIPTION
**Pull request by nobeeakon**
### What
- Summary: Navbar to typescript. Updated the way nav links are being set.

Changes:

- App.js is now App.tsx
- ResponsiveNavBar.jsx is now ResponsiveNavBar.tsx
- now instead of a useEffect that is called and sets the same navbar links in various places we have a function that does it and is passed to the places that need it
- Minor cleanup in various places (removed unused variables, etc.)
- sorted imports: first react, then react-router, then all the MUI ones, and last our own files…this will help us in the future (easier to read and clean), but as is just reordering I preferred to do it now..

### CODE REVIEW

File: taxonomy-editor-frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      taxonomyName: string;
+      branchName: string;
+    }) => {
+      const urlPrefix = `${taxonomyName}/${branchName}/`;
+
+      const newNavLinks = [
+        { url: urlPrefix + "entry", translationKey: "Nodes" },
+        { url: urlPrefix + "search", translationKey: "Search" },
+        { url: urlPrefix + "export", translationKey: "Export" },
+      ];
```
Comment (diivi) [inline taxonomy-editor-frontend/src/App.tsx:46]: Just a reminder to add the error page link once #186 gets merged

File: taxonomy-editor-frontend/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    []
+  );
+
+  return (
+    <ThemeProvider theme={theme}>
+      <CssBaseline />
+      <Router>
+        <ResponsiveAppBar displayedPages={navLinks} />
+        <div>
```
Comment (diivi) [inline taxonomy-editor-frontend/src/App.tsx:58]: Is the `div` tag here just for syntactic sugar? I tried removing it and couldn't notice a difference

Comment (nobeeakon) [reply taxonomy-editor-frontend/src/App.tsx]: done, removed the `div`

**Review by aadarsh-ram — APPROVED** [review]
LGTM @nobeeakon, thanks :)

**Pull request merged**